pkg/crc/config/settings.go

```diff
@@ -56,6 +57,17 @@ func RegisterSettings(cfg *Config) {
 		return ValidateBundlePath(value, GetPreset(cfg))
 	}
 
+	validateSmbSharedDirs := func(value interface{}) (bool, string) {
+		if !cfg.Get(HostNetworkAccess).AsBool() {
+			return false, fmt.Sprintf("%s can only be used with %s set to 'true'",
+				EnableSharedDirs, HostNetworkAccess)
+		}
```

Comment:
Is there any way to restrict this to only the ports needed for file sharing?
The VM having access to all the services running on the host, and also knowing the user login/password does not sound great from a security perspective. We've removed a big part of the isolation between host and guest imo.

Reply:
i don't know if its possible to filter for ports, i guess that will also need changes to `gvisor-tap-vsock` 

maybe we could try adding a `:port` to the NAT table entries setup during daemon start and see if that works!